api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleVoucherEntitlementTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.junit.jupiter.api.Test;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+
+class CycleVoucherEntitlementTest {
+
+    @Test
+    void shouldThrowExceptionWhenListIsNull() {
+        IllegalArgumentException thrown = catchThrowableOfType(() -> new CycleVoucherEntitlement(null), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("List of voucher entitlements must not be null or empty.");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenListIsEmpty() {
+        IllegalArgumentException thrown = catchThrowableOfType(() -> new CycleVoucherEntitlement(emptyList()), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("List of voucher entitlements must not be null or empty.");
+    }
+
+    @Test
+    void shouldCreateTotalValuesFromListOfVoucherEntitlements() {
+        VoucherEntitlement voucherEntitlement1 = VoucherEntitlement.builder()
+                .vouchersForChildrenUnderOne(1)
+                .vouchersForChildrenBetweenOneAndFour(2)
+                .vouchersForPregnancy(3)
+                .voucherValueInPence(100)
+                .build();
+        VoucherEntitlement voucherEntitlement2 = VoucherEntitlement.builder()
+                .vouchersForChildrenUnderOne(2)
+                .vouchersForChildrenBetweenOneAndFour(3)
+                .vouchersForPregnancy(3)
+                .voucherValueInPence(100)
+                .build();
+
+        CycleVoucherEntitlement result = new CycleVoucherEntitlement(asList(voucherEntitlement1, voucherEntitlement2));
+
+        assertThat(result.getVouchersForChildrenUnderOne()).isEqualTo(3);
+        assertThat(result.getVouchersForChildrenBetweenOneAndFour()).isEqualTo(5);
+        assertThat(result.getVouchersForPregnancy()).isEqualTo(6);
+        assertThat(result.getVoucherValueInPence()).isEqualTo(100);
+        assertThat(result.getTotalVoucherEntitlement())
```

Comment:
I think it would be clearer to have the actual values in here (plus line below) rather than a calculation as its clearer that the assertion is then correct.

Reply:
I didn't want to do the values here as they are calculated internally in the VoucherEntitlement constructor.

Reply:
There are two choices really, either don't do the assertion or put the exact value in, having the calculation there does not help, in fact is just confuses the test. I always put the exact number it, its clear and accurate.

Reply:
Replaced with exact values.